Add support for writing Ext TCP Hdr TLV

GregoryVds opened the pull request:
Add support for writing an empty Extended TCP Header TLV.
We would need to add C unit tests at some point, as this case is not covered by the client library.

Commit:
Add support for writing Ext TCP Hdr TLV

Upcoming commit will add support for reading Ext TCP Hdr TLV.

Signed-off-by: Gregory Vander Schueren <gregory.vanderschueren@tessares.net>
Change-Id: I4cad166fa78414371d46b69972f16d674cc22bf8

gdetal commented on convert_util.h:
```diff
@@ -71,6 +71,10 @@ struct convert_opts {
 	/* if CONVERT_F_ERROR is set in flags */
 	uint8_t		error_code;
 
+	/* if CONVERT_F_EXTENDED_TCP_HDR is set in flags
```
You could do something like:

```
uint8_t *options;
size_t n_options;
```

options needs to be malloc'ed and free'd by the caller.

GregoryVds replied:
Done for writing. Will add support for reading in another commit.

gdetal commented on convert_util.c:
```diff
@@ -162,6 +162,23 @@ _convert_write_tlv_error(uint8_t *buff, size_t buff_len,
 
+static ssize_t
+_convert_write_tlv_extended_tcp_hdr(uint8_t *buff, size_t buff_len,
+                                    UNUSED struct convert_opts *opts)
```
remove UNUSED

GregoryVds replied:
Done

gdetal commented on convert_util.c:
```diff
@@ -162,6 +162,23 @@ _convert_write_tlv_error(uint8_t *buff, size_t buff_len,
+	if (buff_len < length)
+		return -1;
+
+	/* TODO support actually announcing TCP options */
```
Here you can go through the opts->options and write them in the TLV.

GregoryVds replied:
Done

gdetal reviewed (changes requested):
Thanks for the change. Can you update the code such that the Convert Header TLV can be filled with TCP options ?

Commit:
_convert_write_tls() now takes const convert_opts

Makes more sense not to modify the struct convert_opts passed by the
called. Just make a local copy of the flags to control the loop.

Signed-off-by: Gregory Vander Schueren <gregory.vanderschueren@tessares.net>
Change-Id: If3a3bcb203b11c5cbe72be2707c2318ab5cccd52

Commit:
Separate libconvert_client and libconvert_util

Rename libconvert to libconvert_client to differentiate from
libconvert_util, which is now compiled as a lib as well.

Signed-off-by: Gregory Vander Schueren <gregory.vanderschueren@tessares.net>
Change-Id: Idde0f6a51e47551df4e736ef3b4841d5d993f664

Commit:
Add infrastructure for unit testing libconvert_util

Signed-off-by: Gregory Vander Schueren <gregory.vanderschueren@tessares.net>
Change-Id: I2362e53ac4761b5c1e4736d086f3389e841be48d
